interface/src/app/gene/genes.js

```diff
@@ -0,0 +1,376 @@
+angular.module('adage.gene.search', [
+  'adage.gene.resource'
+])
+
+.config(function($stateProvider) {
+  $stateProvider
+    .state( 'gene_search', {
+      url: '/gene_search',
+      views: {
+        "main": {
+          templateUrl: 'gene/gene-network.tpl.html',
+          controller: ['$scope', 'UserFactory',
+            function($scope, UserFactory) {
+              UserFactory.getPromise().$promise.then( function() {
+                $scope.userObj = UserFactory.getUser();
+              });
+            }
+          ]
+        }
+      },
+      data: { pageTitle: 'Gene Search' }
+    })
+
+    .state( 'gene_network', {
+      url: '/gene_network/',
+      views: {
+        "main": {
+        }
+      },
+      data: { pageTitle: 'Gene Network' }
+    })
+
+    ;
+})
+
+.factory('SearchResults', ['$rootScope', 'Gene', function( $rootScope, Gene ) {
+    var queries = [];
+    var searchResults = {};
+
+    return {
+        getQueries: function () {
+            return queries;
+        },
+        getQueryResults: function ( query ) {
+            return searchResults[query];
+        },
+        getSearchResults: function () {
+            return searchResults;
+        },
+        remove: function ( query ) { // Remove a query and its associated result
+            queries = queries.filter(function(el) { return el != query; });
+            delete searchResults[ query ];
+            $rootScope.$broadcast( 'results.update' );
+        },
+        clear: function () { // Clear the service
+            queries = [];
+            searchResults = {};
+            $rootScope.$broadcast( 'results.update' );
+        },
+        size: function() {
+            return queries.length;
+        },
+        search: function ( qparams ) {
+          // Search for genes and add the results to the service
+            Gene.search(qparams, function(data) {
+                var previousQueries = queries.length;
+                for (var i = 0; i < data.length; i++) {
+                    query = data[i].search;
+                    if (!searchResults[ query ]) {
+                      //if search term didn't already exist
+                      searchResults[ query ] = data[i]; //add it to the results
+                      queries.push(query);// add to the list of queries
+                    }
+                }
+                if (previousQueries != queries.length) {
+                    $rootScope.$broadcast( 'results.update' );
+                }
+                $rootScope.$broadcast( 'results.searchResultsReturned' );
+            });
+        }
+    };
+}])
+
+.factory( 'SelectedGenesFactory', function( ) {
+    var selected_genes = {};
```

Comment:
Not really, because the selected_genes will hold the gene information as the value for each key (which will be the gene ID and that one will be an integer). I think it will be useful to keep the gene info. so that we can (for example) provide the standard gene name and a tooltip with the full name of the gene in this interface instead of just the gene ID.